docs/public/components/creating-components.md

```diff
@@ -90,23 +89,62 @@ For our compression example is will look like this:
-for your component. Also note that there is no `Start`/`Stop` method. Anything related to lifecycle will be handle
+for your component. 
+
+When defining a component interface we try to use struct or interface from third-party dependencies. Ensuring that using the component interface doesn't require importing any thrid-party dependency.
+
+!!! warning "Interface using a third-party dependency"
+    ```
+    package telemetrydef
+
+    import "github.com/prometheus/client_golang/prometheus"
+
+    // team: agent-shared-components
+
+    // Component is the component type.
+    type Component interface {
+        // RegisterCollector Registers a Collector with the prometheus registry
+        RegisterCollector(c prometheus.Collector)
+    }
+    ```
+
+For the example above every user of the telemetry component would have to import `github.com/prometheus/client_golang/prometheus`. To avoid those cases is best to design your interface mindfully.
+
+TODO: write alternative solution
+
+Also note that there is no `Start`/`Stop` method. Anything related to lifecycle will be handle
 internally by each component (more on this [here TODO]()).
 
 TODO: write the lifecycle part and update the link above
 
 ### The impl folders
 
-You will need a folder per implementation. In those folder you are free to do whatever you want as long as you expose
-one public function to construct your interface.
+The `impl` folder is where the component implementation is written. The component implementation details is up to the developer to design. 
+The only requirement is that there is a public instantiation function called `NewComponent`.
+
+=== ":octicons-file-code-16: comp/compression/fx-zstd/component.go"
+    ```go
+    package zstdimpl
 
-As explained in the [fx](fx.md) documentation, we use dependency injection to provide required parameters. The nomenclature
-is to use a `Requires` and `Provides` structure to get and return value from your constructor.
+    import (
+        // We always import the component def folder to be able to return a 'def.Component' type.
+        "github.com/DataDog/datadog-agent/comp/compression/def"
+    )
 
-=== ":octicons-file-code-16: comp/&lt;bundleName&gt;/compression/fx-zstd/component.go"
+    // NewComponent returns a new ZSTD implementation for the compression component
+    func NewComponent(){
+        ....
+    }
+    ```
+
+To access an argument in the  `NewComponent` function we use a special struct name `Requires`. And the return value is a special stuct named `Provides`. Those are internal nomenclature used to handle the different component depencies using `FX`
+
+For our component we are going to need to access the configuration component and the log component. To express that in our component we are going to define a `Requires` struct with two fields. The name is irrelevant but the type must by a concrete type. 
```

Comment:
This seems more convoluted than the original one liner, no ? We could add a line saying that FX/dependencies injection is optional but not `Requires` and `Provides` ?